x-pack/filebeat/input/httpjson/request.go

```diff
@@ -88,24 +89,44 @@ type requestFactory struct {
 	password   string
 	log        *logp.Logger
 	encoder    encoderFunc
+	replace    string
```

Comment:
Can this be compiled to a `*regexp.Regexp` during construction rather than repeatedly in `generateNewUrl`?

Reply:
Right.